### PR TITLE
Yeet System

### PR DESCRIPTION
**Pull request by anishch**
Vinay and I feel like we have missed something big but cannot remember - other than that, it seems done but we'd like people to review it.

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.EnumMap;
+
+public class YeetSystem {
+
+    private static final double LEFT_CLOSED_POSITION = 0.715;
+    private static final double RIGHT_CLOSED_POSITION = 0.189;
+    private static final double LEFT_OPEN_POSITION = 0.446;
+    private static final double RIGHT_OPEN_POSITION = 0.456;
+
+    public enum Servoes {
```
Comment (zerbash) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java]: Check spelling!

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        private final double closedPosition;
+        private boolean closed;
+
+        Servoes(double closedPosition, double openPosition, boolean closed) {
+            this.closedPosition = closedPosition;
+            this.openPosition = openPosition;
+            this.closed = closed;
+        }
+
+        private double getOpenPosition() {
```
Comment (zerbash) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java]: Should be public

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    private void init() {
+        down();
+        release();
+    }
+
+    private void up(){
+        grab();
+        while (motor.getCurrentPosition() != NUM_REVOLUTIONS * TICKS_PER_REVOLUTION){
```
Comment (zerbash) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java]: Do you need to make these calculations on _every_ iteration?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            motor.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
```
Comment (zerbash) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java]: Do these need to be set every iteration?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    DcMotor motor; //one motor that we need
+
+    private static final double TICKS_PER_REVOLUTION = DriveConstants.TICKS_PER_REV; //number of ticks per revolution
+    private static final double NUM_REVOLUTIONS = 0.8; // this needs to be changed - the number is num of revolutions
+
+
+    private static final int DEFAULT = 0; // this needs to be changed
+
+
+    public YeetSystem(DcMotor motor1, EnumMap<Servoes, Servo> servoMap) { //constructor
```
Comment (zerbash) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java]: is there a motor2?

Comment (anishch) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/YeetSystem.java]: There used to be; I'll go ahead and just change the name.

**Pull request merged**